druid/src/widget/svg.rs

```diff
@@ -150,20 +156,24 @@ impl SvgData {
                                 x,
                                 y,
                             } => {
-                                let x1 = (x1 * scale) + offset.x;
-                                let y1 = (y1 * scale) + offset.y;
-                                let x2 = (x2 * scale) + offset.x;
-                                let y2 = (y2 * scale) + offset.y;
-                                let x = (x * scale) + offset.x;
-                                let y = (y * scale) + offset.y;
-
                                 path.curve_to((x1, y1), (x2, y2), (x, y));
                             }
                             usvg::PathSegment::ClosePath => {
                                 path.close_path();
                             }
                         }
                     }
+
+                    path.apply_affine(Affine::new([
+                        p.transform.a,
+                        p.transform.b,
+                        p.transform.c,
+                        p.transform.d,
+                        p.transform.e,
+                        p.transform.f,
+                    ]));
+                    path.apply_affine(offset_matrix);
```

Comment:
i'm really scratching my head here. where does `offset_matrix` come from?

Reply:
The second commit applies the transform matrix but it also refractors the old offset and scale transforms to happen after the transform matrix has been applied as matrix multiplication is not like scaler multiplication so needs to happen in the right order, this seemed like a much more succinct way of doing it.

maybe scale_offset_matrix may be better?

I think i could fudge the maths so that i don't need to move the old scale and offset transforms and instead apply a matrix to the transform matrix before applying it bu this seems much nicer.

Reply:
oh i see what you mean! i had messed up a rebase to fix some other comments :( my very bad!